docs/contract.yaml

```diff
@@ -0,0 +1,156 @@
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/OperationSuccess"
+        '406':
+          description: return information relevant to the conflict in case of a failed union
+          content:
+            application/json:
+              schema:
+                type: object
+                properties:
+                  conflicts:
+                    type: array
+                    items:
+                      "$ref": "#/components/schemas/MergeConflictObject"
+        default:
+          description: error
+          content:
+            application/json:
+              schema:
+                "$ref":  "#/components/schemas/Error"  
+
+  /specMath/overlay:
+    post:
+      description: |
+        the overlay endpoint takes a spec and an overlay file (which is a partial spec)
+        and performs a specialized union where the overlay takes priority. In case of a collision, a
+        conflict will not be reported and the value from the overlay will be taken.
+      operationId: specMathOverlay
+      requestBody:
+        description: data needed for the overlay operation
+        required: true
+        content: 
+          application/json:
+            schema: 
+              "$ref": "#/components/schemas/OverlayBody"
+      responses:
+        '200':
+          description: operation was successful, returns the resultant spec 
+          content:
+            application/json:
+              schema:
+                "$ref": "#/components/schemas/OperationSuccess"   
+        default:
+          description: error
+          content:
+            application/json:
+              schema:
+                "$ref":  "#/components/schemas/Error"                
+components:
+  schemas:
+    MergeBody:
+      description: holds data required for the union operation
+      type: object
+      required:
+        - spec1
+        - spec2
+      properties:
+        spec1:
+          description: an OpenAPI specification
+          type: string
+        spec2:
+          description: an OpenAPI specification
+          type: string
+        defaults:
+          description: an OpenAPI specification fragment which contains metadata about the merged spec
+          type: string
+        conflictResolutions:
+          description: an array of MergeConflictObjects which specify resolutions to conflicting keypaths
+          type: array
+          items:
+            "$ref": "#/components/schemas/MergeConflictObject"
+    
+    OverlayBody:
+      description: holds data required for the overlay operation
+      type: object
+      required:
+        - spec
+        - overlay
+      properties:
+        spec:
+          description: an OpenAPI specification
+          type: string
+        overlay:
+          description: an OpenAPI specification fragment which contains metadata about the merged spec
+          type: string
+
+    MergeConflictObject:
```

Comment:
I think the "Object" suffix should be dropped. It's superfluous and also conflicts with the way the "MergeBody" schema is named (with no "Object" suffix).